src/internal/operations/backup.go

```diff
@@ -425,7 +426,8 @@ func consumeBackupCollections(
 		kopia.TagBackupCategory: "",
 	}
 
-	for _, reason := range reasons {
+	reunion := append(slices.Clone(reasons), fallbackReasons...)
+	for _, reason := range reunion {
```

Comment:
Doh, I keep forgetting about the `and`.  Going to close this, in that case, as I don't think we have any additional work to do here.

Reply:
@ashmrtn Wait a second: double checking.  This isn't a lookup, it's attaching the tags to the snapshot.  Kopia only assets tag intersection on lookup.  iirc, we're not required to match _all_ tags on the manifest; rather, all tags that we request must be present on the manifest in order to fetch it.  So it's okay for us to add the fallbackReasons here, as that acts as a superset which won't interfere with looking up the tags from `reasons` in the future.

Whether or not we need this at all is another question.  Mostly wanting to make sure I have ^ grokked correctly.  We would be safe to make this change, right?

Reply:
>all tags that we request must be present on the manifest in order to fetch it

Hmm that's true

>Whether or not we need this at all is another question

I don't think it's useful for us to tag snapshots with both the PN and the ID of things. We're moving away from PN for a reason, so it would be good not to continue to have it floating around. Adding PN would also mean that snapshots would match both the fallback and "main" reasons we use which means we'd have 2x the data to wade through

What we _do_ need to ensure we test prior to switching things over is that the subtree path for the IncrementalBase is populated according to whether the base snapshot has PN or ID. I know the code looks like it should just "do the right thing" but would really feel better to have more hard results of some form on this

Reply:
Yup, that sort of testing is my priority today.